File: project/donghyeon-onboarding/src/main/java/me/dhlee/donghyeononboarding/common/ApiResponse.java
package me.dhlee.donghyeononboarding.common;

public record ApiResponse<T>(
    int httpCode,
    String message,
    T data
) {

    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(200, null, data);
    }

    public static <T> ApiResponse<T> badRequest(String message) {
        return new ApiResponse<>(400, message, null);
    }
}

File: project/kyungtae-onboarding/survey-api/src/main/kotlin/com/innercircle/api/common/response/ApiResponse.kt
package com.innercircle.api.common.response

data class ApiResponse<T>(
    val message: String? = null,
    val data: T? = null,
    val code: ApiResponseCode = ApiResponseCode.SUCCESS
) {

    companion object {
        fun <T> created(data: T): ApiResponse<T> {
            return ApiResponse(data = data)
        }

        fun void(): ApiResponse<Void> {
            return ApiResponse()
        }

        fun <T> error(code: ApiResponseCode): ApiResponse<T> {
            return ApiResponse(message = code.message, code = code)
        }

        fun <T> ok(data: T): ApiResponse<T> {
            return ApiResponse(data = data)
        }
    }
}
